tests/unittests/test-cerrorinfo.cpp

```diff
@@ -0,0 +1,319 @@
+	warranty. In no event will the authors be held liable for any damages
+	arising from the use of this software.
+
+	Permission is granted to anyone to use this software for any purpose,
+	including commercial applications, and to alter it and redistribute it
+	freely, subject to the following restrictions:
+
+		1. The origin of this software must not be misrepresented;
+		   you must not claim that you wrote the original software.
+		   If you use this software in a product, an acknowledgment
+		   in the product documentation would be appreciated but is
+		   not required.
+
+		2. Altered source versions must be plainly marked as such,
+		   and must not be misrepresented as being the original software.
+
+		3. This notice may not be removed or altered from any source
+		   distribution.
+*/
+#include <gtest/gtest.h>
+
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif /* #ifndef NOMINMAX */
+
+#include <tchar.h>
+#include <Windows.h>
+
+#include <wrl.h>
+
+#include "basis/CErrorInfo.h"
+
+/*!
+ * @brief テンプレートクラスの機能確認
+ */
+TEST(TComImpl, AddRef_Release)
+{
+	// IErrorInfoを生成する
+	IErrorInfo* pErrorInfo(reinterpret_cast<IErrorInfo*>(MakeMsgError(L"test")));
+
+	// 初期値は0なので、AddRefすると1が返る
+	ASSERT_EQ(1, pErrorInfo->AddRef());
+
+	// 内部カウンタは1なので、Releaseすると0になる
+	ASSERT_EQ(0, pErrorInfo->Release());
+}
+
+/*!
+ * @brief テンプレートクラスの機能確認
+ */
+TEST(TComImpl, QueryInterface_BadPointer)
+{
+	// IErrorInfoを生成する
+	IErrorInfo* pErrorInfo(reinterpret_cast<IErrorInfo*>(MakeMsgError(L"test")));
+
+	// 初期値は0なので、AddRefすると1が返る
+	ASSERT_EQ(1, pErrorInfo->AddRef());
+
+	// QueryInterfaceのOutポインタにnullptrを指定するとポインタエラーになる
+	ASSERT_EQ(E_POINTER, pErrorInfo->QueryInterface(GUID_NULL, nullptr));
+
+	// 内部カウンタは1なので、Releaseすると0になる
+	ASSERT_EQ(0, pErrorInfo->Release());
+}
+
+/*!
+ * @brief テンプレートクラスの機能確認
+ */
+TEST(TComImpl, QueryInterface_IUnknown)
+{
+	using namespace Microsoft::WRL;
+
+	// IErrorInfoを生成する
+	IErrorInfo* pErrorInfo(reinterpret_cast<IErrorInfo*>(MakeMsgError(L"test")));
+
+	// 初期値は0なので、AddRefすると1が返る
+	ASSERT_EQ(1, pErrorInfo->AddRef());
+
+	// IUnknownを取得する
+	ComPtr<IUnknown> pUnknown;
+	ASSERT_TRUE(SUCCEEDED(pErrorInfo->QueryInterface(pUnknown.GetAddressOf())));
+
+	//pUnknownを解放する
+	pUnknown = nullptr;
+
+	// 内部カウンタは1に戻っているはずなので、Releaseすると0になる
+	ASSERT_EQ(0, pErrorInfo->Release());
+}
+
+/*!
+ * @brief テンプレートクラスの機能確認
+ */
+TEST(TComImpl, QueryInterface_Imlemented)
+{
+	using namespace Microsoft::WRL;
+
+	// IErrorInfoを生成する
+	IErrorInfo* pErrorInfo(reinterpret_cast<IErrorInfo*>(MakeMsgError(L"test")));
+
+	// 初期値は0なので、AddRefすると1が返る
+	ASSERT_EQ(1, pErrorInfo->AddRef());
+
+	// IErrorInfoを取得する
+	ComPtr<IErrorInfo> pOther;
+	ASSERT_TRUE(SUCCEEDED(pErrorInfo->QueryInterface(pOther.GetAddressOf())));
+
+	//pOtherを解放する
+	pOther = nullptr;
+
+	// 内部カウンタは1に戻っているはずなので、Releaseすると0になる
+	ASSERT_EQ(0, pErrorInfo->Release());
+}
+
+/*!
+ * @brief テンプレートクラスの機能確認
+ */
+TEST(TComImpl, QueryInterface_NotImlemented)
```

Comment:
```suggestion
TEST(TComImpl, QueryInterface_NotImplemented)
```
コピりましたね。こちらもスペル自信ないです。